drivers/dai/intel/dmic/dmic.c

```diff
@@ -174,30 +250,53 @@ static void dai_dmic_irq_handler(const void *data)
 
 static inline void dai_dmic_dis_clk_gating(const struct dai_intel_dmic *dmic)
 {
+#ifdef CONFIG_SOC_SERIES_INTEL_CAVS_V15
```

Comment:
I think it would be good to have shim registers definitions and basic access functions separated from main DMIC driver code and moved to platform specific headers. This patch now adds #ifdef CONFIG_SOC_SERIES_INTEL_CAVS_V15, but shim registers definition for SOC_SERIES_INTEL_ACE will change from one version to another.  In result soon we would have here several #ifdefs  for different generations.

Reply:
@juimonen @jxstelter Is this something to do now, or later in separate PR. Given we only have two cases now, I'd prefer to do this in separate PR.

Reply:
OK.  Separate PR will be good.